src/renderer/src/utils/api.ts

```diff
@@ -8,31 +12,141 @@ export function formatApiKeys(value: string): string {
   return value.replaceAll('，', ',').replaceAll('\n', ',')
 }
 
+/**
+ * 判断 host 的 path 中是否包含形如版本的字符串（例如 /v1、/v2beta 等），
+ *
+ * @param host - 要检查的 host 或 path 字符串
+ * @returns 如果 path 中包含版本字符串则返回 true，否则 false
+ */
+export function hasAPIVersion(host?: string): boolean {
+  if (!host) return false
+
+  const versionRegex = /\/v\d+([a-z0-9._-]*)/i
+
+  try {
+    const url = new URL(host)
+    return versionRegex.test(url.pathname)
+  } catch {
+    // 若无法作为完整 URL 解析，则当作路径直接检测
+    return versionRegex.test(host)
+  }
+}
+
+export function withoutTrailingSlash(url: string | undefined) {
+  return url?.replace(/(\/|:)$/, '')
+}
```

Comment:
Use generics to obtain a precise return type

```suggestion
export function withoutTrailingSlash<T extends string | undefined>(url: T): T {
  return url?.replace(/(\/|:)$/, '') as T
}
```